torchvision/prototype/datasets/_builtin/stanford_cars.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Dict, List, Tuple, Iterator
+
+import numpy as np
+from torchdata.datapipes.iter import Filter, IterDataPipe, Mapper, Zipper
+from torchvision.prototype.datasets.utils import Dataset, DatasetConfig, DatasetInfo, HttpResource
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling, path_comparator, read_mat
+from torchvision.prototype.features import BoundingBox, EncodedImage
+
+
+class _StanfordCarsLabelReader(IterDataPipe[Tuple[np.ndarray, int]]):
+    def __init__(self, datapipe: IterDataPipe[Dict[str, Any]]) -> None:
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[Tuple[str]]:
+        for _, file in self.datapipe:
+            file = iter(read_mat(file, squeeze_me=True)["annotations"])
+            for line in file:
+                yield line
```

Comment:
This was hard to parse for me until I run this code myself. Maybe something like this?

```suggestion
    def __iter__(self) -> Iterator[Tuple[int, int, int, int, int, str]]:
        for _, file in self.datapipe:
            data = read_mat(file, squeeze_me=True)
            for ann in data["annotations"]:
                yield tuple(ann)
```

Reply:
Oh sorry! My bad! I should have made it more readable.